pkg/store/bucket.go

```diff
@@ -772,6 +773,75 @@ func blockSeries(
 	return newBucketSeriesSet(res), indexr.stats.merge(chunkr.stats), nil
 }
 
+// blockSeriesWithoutChunks is the same as blockSeries function except it only
+// fetches labels and doesn't touch chunks. This is used in /api/v1/series API.
+func blockSeriesWithoutChunks(
```

Comment:
Do we need to duplicate `blockSeries()`? How more dirty would be `blockSeries()` code if we add a "skipChunks" parameter to that function? Or could we make it implicitly skipping chunks if `chunkr` is `nil`?

Reply:
Thanks, let me try

Reply:
Done. Please take a look!